aws-lambda/README.md

```diff
@@ -0,0 +1,281 @@
+Demo requirements: 
+* [AWS Serverless Application Model (AWS SAM)](https://aws.amazon.com/serverless/sam/)
+* [Docker](https://docs.docker.com/get-docker/)
+
+Clone the repository and change into the demo directory
+```
+git clone https://github.com/aws/aws-graviton-getting-started
+cd aws-lambda/GravitonLambdaNumber
+```
+### Migrating a Lambda function from x86 to arm64
+This demo shows how to migrate an existing Lambda function from x86 to arm64.
+The Node.js function code in [`/src/app.js`](/src/app.js) uses the [axios](https://www.npmjs.com/package/axios) client to connect to a third part service, [http://numbersapi.com/](http://numbersapi.com/), to find interesting facts about numbers. As the axios library is not a binary file, it can seamlessly work on Graviton2 without compilation.
+
+The existing application consists of an API endpoint which invokes the Lambda function, retrieves the number fact, and returns the response.
+
+Build the existing x86 function version using AWS SAM.
+```
+sam build
+```
+![sam build](/aws-lambda/img/sambuild.png)
+
+Deploy the function to your AWS account:
+```
+sam deploy --stack-name GravitonLambdaNumber -g
+```
+Accept the initial defaults, and ensure you enter Y for `LambdaNumberFunction may not have authorization defined, Is this okay? [y/N]: y`
+
+![sam deploy -g](/aws-lambda/img/samdeploy-g.png)
+
+AWS SAM deploys the infrastructure and outputs an APIBasePath
+
+![ApiBasePath](/aws-lambda/img/ApiBasePath.png)
+
+Use `curl` and make a POST request to the APIBasePath URL with a number as a date.
+```
+curl -X POST https://6ioqy4z9ee.execute-api.us-east-1.amazonaws.com -H 'Content-Type: application/json' -d '{ "number": "345", "type": "date" }'
+```
+The Lambda function should respond with the `x64` processor architecture and a fact about the date.
+
+![curl x86](/aws-lambda/img/curlx86.png)
+
+Amend the processor architecture within the AWS SAM [template.yml](./GravitonLambdaNumber/template.yml) file.
+replace 
+```
+Architectures:
+- x86_64
+```        
+with
+```
+Architectures:
+- arm64
+```        
+
+Build the function using an arm64 architecture and deploy the change to the cloud.
+```
+sam build
+sam deploy
+```
+Use the same `curl` command, making a POST request to the APIBasePath URL with a number as a date.
+```
+curl -X POST https://6ioqy4z9ee.execute-api.us-east-1.amazonaws.com -H 'Content-Type: application/json' -d '{ "number": "345", "type": "date" }'
+```
+The Lambda function should respond with the `arm64` processor architecture and a fact about the date.
+
+![curl arm64](/aws-lambda/img/curlarm64.png)
+
+The function has seamlessly migrated from x86 to arm64.
+
+### Building a Lambda function with binary dependencies
+
+This function has no binary dependencies. If you do have a function that required compilation for arm64, AWS SAM can use an arm64 build container image to create the artifacts for arm64.
+This allows you to build arm64 specific dependencies on an x86 workstation.
```

Comment:
does it work on an M1 mac too?

Reply:
Amended wording to specify it works in either direction.